### PR TITLE
Iaas networking service endpoint support

### LINKED ISSUE #276: New-CmAzIaasNetworking - Support for Service Endpoints
We need to support Service Endpoints in a non-disruptive deployment, per requirement raised in #205.

An [example template](https://github.com/Azure/azure-quickstart-templates/blob/master/201-vnet-2subnets-service-endpoints-storage-integration/azuredeploy.json)) shows how this works at an ARM level.

This needs to be added to the _current networking stack_, but as Eliot points out in the linked issue we'll need to investigate whether a redeploy would remove the subnet and service endpoint along with it for a period of time. We want to minimise disruption as much as possible. Handling service endpoints within the networking stack seems to have the best chance of achieving this.

For each type of input:

 - **CSV**
    Add a new column to the CSV file (to the end, assuming that makes sense) containing the list of service endpoint services, which can be pipe-separated.

 - **YML**
   Use the nested structure with an array of services to add endpoints for.

### PR DESCRIPTION
**Pull request by deepend-dev**
closes #276 by providing service endpoint support in both csv and yml versions

### CODE REVIEW

**Review by tm-luke — CHANGES_REQUESTED** [review]

File: azure-integration-pipelines.yml [hunk 1]
```diff
@@ -108,13 +108,17 @@ stages:
...
+          Write-Host "Testing the yml deployment..."
           New-CmAzIaasNetworking `
             -VnetsCsvFile "$(Build.SourcesDirectory)/$(moduleName)/Resources/Project/Integration/networking/csv/vnet.csv" `
             -RouteTablesCsvFile "$(Build.SourcesDirectory)/$(moduleName)/Resources/Project/Integration/networking/csv/udr.csv" `
             -NsgsCsvFile "$(Build.SourcesDirectory)/$(moduleName)/Resources/Project/Integration/networking/csv/nsg.csv" `
             -Verbose `
             -Confirm:$false
 
+          Write-Host "Testing the yml deployment by central deployment script..."
+          New-CmAzDeployment -SettingsFile "$(Build.SourcesDirectory)/$(moduleName)/Resources/Project/Integration/networking/yml/example/networking.yml" -Verbose -Confirm:$false
```
[the remark this answers is not in the document]
Comment (deepend-dev) [reply azure-integration-pipelines.yml:120]: done

File: Cloudmarque.Azure/Resources/Project/Integration/networking/yml/example/vnets/Iaas-Test-Spoke1-Vnet1.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+vnetName: Iaas-Test-Spoke1-Vnet1
+vnetpeerings:
+addressSpace:
+  - 10.10.0.0/22
+dnsServers: ''
```
[the remark this answers is not in the document]
Comment (deepend-dev) [reply Cloudmarque.Azure/Resources/Project/Integration/networking/yml/example/vnets/Iaas-Test-Spoke1-Vnet1.yml:5]: no we donot. Its all autogenerated

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/managedNetwork/New-CmAzIaasNetworking.ps1 [hunk 3]
```diff
@@ -735,12 +764,35 @@
...
 
-							$vnetObjectYml.subnets | Where-Object { !$_.routeTable } | ForEach-Object {
-								$_.routeTable = ""
+										if ($allowedServiceEndpoints.Contains($endpoint)) {
+
+											$buildServiceEndpoints += @{ service = "Microsoft.$endpoint" }
+										}
+										else {
+
+											if ($endpoint -ne '') {
```
[the remark this answers is not in the document]
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/IaaS/networking/managedNetwork/New-CmAzIaasNetworking.ps1:788]: ok will do

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/managedNetwork/New-CmAzIaasNetworking.ps1 [hunk 4]
```diff
@@ -191,12 +197,35 @@
...
+
+							$endpoint = @{
+								service = "Microsoft.$_"
+							}
+
+							$subnetServiceEndpoints += $endpoint
+						}
+						else {
+
+							if ($_ -ne '') {
```
[the remark this answers is not in the document]
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/IaaS/networking/managedNetwork/New-CmAzIaasNetworking.ps1:214]: done

**Review by tm-luke — APPROVED** [review]

**Pull request merged**